### ISSUE TITLE
[operator] bump operator base image to 4.17 / 1.35.0

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#7856]
The production ansible operator base image is going up to 4.17 - we need to move to it. Do what we did last time when we moved to 4.15: https://github.com/kiali/kiali/issues/7385

Note that the "upstream 4.x" images (what I called the "upstream-downstream" image) is no longer being produced. So 4.x images here will no longer be updated: https://quay.io/repository/openshift/origin-ansible-operator?tab=tags (this is due to [this PR](https://github.com/openshift/release/pull/49084)).  For the Kiali project, we must use the upstream images found in https://quay.io/repository/operator-framework/ansible-operator (the 1.x line of upstream ansible operator).

Note also the change in downstream image name - there is now `-rhel9-` in the name.

* Downstream v4.17: https://catalog.redhat.com/software/containers/openshift4/ose-ansible-rhel9-operator/6527e49eadb44b09d465a57b
* Upstream 1.35.0: https://quay.io/repository/operator-framework/ansible-operator?tab=tags&tag=v1.35.0

4.17 should correspond to ansible operator SDK 1.35.0

Downstream 4.17:
```bash
$ podman run -it --rm registry.redhat.io/openshift4/ose-ansible-operator:v4.17 | head -n 1
{"level":"info","ts":"2024-10-23T14:52:47Z","logger":"cmd","msg":"Version","Go Version":"go1.22.5 (Red Hat 1.22.5-1.module+el8.10.0+22070+9237f38b) X:strictfipsruntime","GOOS":"linux","GOARCH":"amd64","ansible-operator":"v4.17.0-202408270335.p0.g6d23192.assembly.stream.el8-dirty","commit":"79a84b30524859899cc89bf0067e1c4bbc4c79a8"}
```

(Currently, I do not know a way to determine what upstream version of ansible operator is inside the downstream one - the log messages are different now. _UPDATE_: this will be fixed in the next z-stream release of the 4.17 image - see [OCPBUGS-43759](https://issues.redhat.com/browse/OCPBUGS-43759)).

The SDK release notes: https://github.com/operator-framework/operator-sdk/releases

**Comment by jmazzitelli** [comment]
## Ansible and Python

Ansible 2.15 with Python 3.9

If you want this version of ansible on your local dev machine, you can clone that tag via: `git clone -b v2.15.12 --depth 1 https://github.com/ansible/ansible.git`

Downstream 4.17:
```
$ podman run  -it --rm --entrypoint '' registry.redhat.io/openshift4/ose-ansible-rhel9-operator:v4.17 ansible --version
ansible [core 2.15.12]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/openshift']
  ansible python module location = /usr/local/lib/python3.9/site-packages/ansible
  ansible collection location = /opt/ansible/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.18 (main, Aug 23 2024, 00:00:00) [GCC 11.4.1 20231218 (Red Hat 11.4.1-3)] (/usr/bin/python3)
  jinja version = 3.1.4
  libyaml = False
```

Upstream 1.35.0:
```
$ podman run  -it --rm --entrypoint '' quay.io/operator-framework/ansible-operator:v1.35.0 ansible --version
ansible [core 2.15.11]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/openshift']
  ansible python module location = /usr/local/lib/python3.9/site-packages/ansible
  ansible collection location = /opt/ansible/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.19 (main, May 16 2024, 08:45:40) [GCC 8.5.0 20210514 (Red Hat 8.5.0-22)] (/usr/bin/python3)
  jinja version = 3.1.4
  libyaml = True
```

**Comment by jmazzitelli** [comment]
## Ansible Collections

_NOTE! Due to a bug in kubernetes.core collection, we needed to upgrade to 4.0.0 previously. We must stay at least on 4.0.0. See https://github.com/kiali/kiali/issues/7476_

There are no collections in the upstream image (run this to see: `podman run -it --rm --entrypoint '' quay.io/operator-framework/ansible-operator:v1.35.0 ansible-galaxy collection list`)

Collections we care about the most:
- kubernetes.core = 2.4.2
- operator_sdk.util = 0.5.0

```sh
podman run -it --rm --entrypoint '' registry.redhat.io/openshift4/ose-ansible-rhel9-operator:v4.17 ansible-galaxy collection list
```

```
# /opt/ansible/.ansible/collections/ansible_collections
Collection                               Version
---------------------------------------- -------
cloud.common                             2.1.1  
community.docker                         3.10.3 
community.library_inventory_filtering_v1 1.0.1  
kubernetes.core                          2.4.2  
operator_sdk.util                        0.5.0  
```

**Comment by jmazzitelli** [comment]
## Python Libraries

query.sh:
```sh
for i in \
  registry.redhat.io/openshift4/ose-ansible-rhel9-operator:v4.17 \
  quay.io/operator-framework/ansible-operator:v1.35.0
do
  filename="/tmp/$(echo $i | sed 's#/#-#g').OUTPUT"
  echo "=== IMAGE: ${i} -> ${filename} ==="
  podman run --rm -i -t --entrypoint "" "${i}" pip3 list --format columns &> $filename
  cat $filename
done
```

That script will output files you can diff to see the differences between images or grep for things. Here's the full output of everything:

```
=== IMAGE: registry.redhat.io/openshift4/ose-ansible-rhel9-operator:v4.17 -> /tmp/registry.redhat.io-openshift4-ose-ansible-rhel9-operator:v4.17.OUTPUT ===
Package             Version
------------------- -----------
ansible-core        2.15.12
ansible-runner      2.3.6
ansible-runner-http 1.0.0
cachetools          5.5.0
certifi             2024.8.30
cffi                1.14.5
chardet             4.0.0
charset-normalizer  3.3.2
cryptography        36.0.1
Cython              3.0.11
dbus-python         1.2.18
decorator           4.4.2
docutils            0.21.2
flit_core           3.9.0
google-auth         2.35.0
gpg                 1.15.1
idna                3.10
importlib_metadata  6.2.1
importlib_resources 5.0.7
iniparse            0.4
Jinja2              3.1.4
kubernetes          29.0.0
libcomps            0.1.18
lockfile            0.12.2
MarkupSafe          2.1.5
oauthlib            3.2.2
packaging           24.1
pbr                 6.1.0
pexpect             4.9.0
pip                 21.2.3
ply                 3.11
poetry-core         1.9.0
ptyprocess          0.7.0
pyasn1              0.6.1
pyasn1_modules      0.4.1
pycparser           2.20
PyGObject           3.40.1
pyinotify           0.9.6
PySocks             1.7.1
python-daemon       3.0.1
python-dateutil     2.9.0.post0
PyYAML              6.0.2
requests            2.31.0
requests-oauthlib   2.0.0
requests-unixsocket 0.3.0
resolvelib          1.0.1
rpm                 4.16.1.3
rsa                 4.9
semantic_version    2.10.0
setuptools          70.0.0
setuptools-rust     1.10.1
setuptools_scm      7.1.0
six                 1.16.0
systemd-python      234
tomli               2.0.1
typing_extensions   4.12.2
urllib3             1.26.20
websocket-client    1.8.0
wheel               0.44.0
zipp                3.20.2
=== IMAGE: quay.io/operator-framework/ansible-operator:v1.35.0 -> /tmp/quay.io-operator-framework-ansible-operator:v1.35.0.OUTPUT ===
Package             Version
------------------- -----------
ansible-core        2.15.11
ansible-runner      2.3.6
ansible-runner-http 1.0.0
cachetools          5.3.3
certifi             2024.2.2
cffi                1.16.0
charset-normalizer  3.3.2
cryptography        42.0.7
distlib             0.3.8
docutils            0.21.2
filelock            3.15.4
google-auth         2.29.0
idna                3.7
importlib-metadata  6.2.1
importlib-resources 5.0.7
Jinja2              3.1.4
kubernetes          29.0.0
lockfile            0.12.2
MarkupSafe          2.1.5
oauthlib            3.2.2
packaging           24.0
pexpect             4.9.0
pip                 23.3.2
pipenv              2023.11.15
platformdirs        4.2.2
ptyprocess          0.7.0
pyasn1              0.6.0
pyasn1_modules      0.4.0
pycparser           2.22
python-daemon       3.0.1
python-dateutil     2.9.0.post0
PyYAML              6.0.1
requests            2.31.0
requests-oauthlib   2.0.0
requests-unixsocket 0.3.0
resolvelib          1.0.1
rsa                 4.9
setuptools          69.5.1
six                 1.16.0
urllib3             1.26.18
virtualenv          20.26.3
websocket-client    1.8.0
zipp                3.18.1

[notice] A new release of pip is available: 23.3.2 -> 24.2
[notice] To update, run: python3.9 -m pip install --upgrade pip
```

Here's the kubernetes libraries used for both of them:

```sh
$ grep kubernetes /tmp/*.OUTPUT
/tmp/quay.io-operator-framework-ansible-operator:v1.35.0.OUTPUT:kubernetes          29.0.0
/tmp/registry.redhat.io-openshift4-ose-ansible-rhel9-operator:v4.17.OUTPUT:kubernetes          29.0.0
```

**Comment by jmazzitelli** [comment]
I did a quick test on minikube. I changed to using downstream and it worked.

```sh
make -e CLUSTER_TYPE=minikube  \
  OPERATOR_BASE_IMAGE_REPO=registry.redhat.io/openshift4/ose-ansible-rhel9-operator \
  OPERATOR_BASE_IMAGE_VERSION=v4.17 \
  cluster-push operator-create kiali-create
```
